pentasharp/Data/Configurations/UserConfiguration.cs

```diff
@@ -0,0 +1,45 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using pentasharp.Models.Entities;
+
+namespace pentasharp.Data.Configurations
+{
```

Comment:
Use FluentAPI with separate methods for clarity, meaning if your configurations grow separating them into dedicated methods can be more readable.
We can have 
    private void ConfigureKeys(EntityTypeBuilder<User> builder)
    {
        builder.HasKey(u => u.UserId);
    }
then  private void ConfigureProperties(EntityTypeBuilder<User> builder)
    {
        builder.Property(u => u.UserId)
            .ValueGeneratedOnAdd();

        builder.Property(u => u.FirstName)
            .IsRequired()
            .HasMaxLength(100);
            .
            .
            .
            } and so on